### PR TITLE
Option Equality

### PR DESCRIPTION
**Pull request by ChrisWallace2020**
`__eq__` function in `BaseOption` that checks equality between all attributes. All attributes are either Python primitive types, other options, or DataLabeler objects, so hence all have `__eq__` functions already, although an enhancement was made to the labeler `__eq__` in this PR to better handle Nones, before it would error.

### CODE REVIEW

File: dataprofiler/profilers/profiler_options.py [hunk 1]
```diff
@@ -110,6 +110,15 @@ def validate(self, raise_error=True):
         elif errors: 
             return errors
 
+    def __eq__(self, other):
+        self_dict = self.__dict__
+        other_dict = other.__dict__
+
+        if type(self) != type(other):
+            return False
```
Comment (ChrisWallace2020) [inline dataprofiler/profilers/profiler_options.py:118]: This ensures the dicts have the same keys in the all statement below

File: dataprofiler/profilers/profiler_options.py [hunk 2]
```diff
@@ -583,6 +592,8 @@ def __init__(self):
         :vartype data_labeler_dirpath: str
         :ivar max_sample_size: Int to decide sample size
         :vartype max_sample_size: int
+        :ivar data_labeler_object: DataLabeler object used in profiler
+        :vartype max_sample_size: BaseDataLabeler
```
Comment (ChrisWallace2020) [inline dataprofiler/profilers/profiler_options.py:596]: Missing docstring I noticed

File: dataprofiler/labelers/base_data_labeler.py [hunk 3]
```diff
@@ -58,6 +58,8 @@ def __eq__(self, other):
         :return: Whether or not self and other are equal
         :rtype: Bool
         """
+        if other is None:
+            return False
```
Comment (ChrisWallace2020) [inline dataprofiler/labelers/base_data_labeler.py:62]: None catching in `BaseDataLabeler.__eq__`

File: dataprofiler/tests/profilers/profiler_options/test_numerical_options.py [hunk 4]
```diff
@@ -1,5 +1,4 @@
-from dataprofiler.profilers.profiler_options \
-     import BooleanOption, NumericalOptions
+from dataprofiler.profilers.profiler_options import NumericalOptions
```
Comment (ChrisWallace2020) [inline dataprofiler/tests/profilers/profiler_options/test_numerical_options.py:1]: Unused import

File: dataprofiler/labelers/base_data_labeler.py (hunk 3 above)
Comment (JGSweets) [reply dataprofiler/labelers/base_data_labeler.py:62]: should this instead be an `isinstance` check?

File: dataprofiler/profilers/profiler_options.py [hunk 5]
```diff
@@ -110,6 +110,15 @@ def validate(self, raise_error=True):
         elif errors: 
             return errors
 
+    def __eq__(self, other):
```
Comment (JGSweets) [inline dataprofiler/profilers/profiler_options.py:113]: add description?

File: dataprofiler/labelers/base_data_labeler.py (hunk 3 above)
Comment (ChrisWallace2020) [reply dataprofiler/labelers/base_data_labeler.py]: Changed it

**Review by gme5078 — APPROVED** [review]
LGTM

**Review by JGSweets — DISMISSED** [review]

**Review by gme5078 — DISMISSED** [review]

**Review by gme5078 — APPROVED** [review]

**Pull request merged**